apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTest.cs

```diff
@@ -85,31 +85,46 @@ private async Task RunTestCase(InstructionList instructionList, Method method, b
     {
         var context = CreateTestContext(method);
         var response = await CreateRetryTestResource(instructionList, method);
-        try
+
+        if (expectSuccess)
+        {
+            RunRetryTest(response, context, method.Group, specifyPreconditions);
+            var postTestResponse = await GetRetryTest(response.Id);
+            Assert.True(postTestResponse.Completed, "Expected retry test completed to be true, but was false.");
+        }
+        else
         {
-            if (expectSuccess)
+            try
             {
                 RunRetryTest(response, context, method.Group, specifyPreconditions);
-                var postTestResponse = await GetRetryTest(response.Id);
-                Assert.True(postTestResponse.Completed, "Expected retry test completed to be true, but was false.");
-            }
-            else
-            {
-                try
-                {
-                    RunRetryTest(response, context, method.Group, specifyPreconditions);
 
                     // storage.buckets.setIamPolicy has no preconditions implemented in .NET and hence will retry successfully
                     if (!method.Name.Contains("buckets.setIamPolicy"))
                     {
                         Assert.False(response.Completed);
                     }
                 }
-                catch (GoogleApiException ex) when (InstructionContainsErrorCode(ex.HttpStatusCode))
+                catch (Exception ex) // To catch expected exception when retry should not happen.
                 {
-                    // The instructions specified that the given status code would be returned.
-                    // We just need to check that we weren't expecting this specific call to succeed.
-                    Assert.False(expectSuccess);
+                    if (ex is GoogleApiException exception)
+                    {
+                        var statusCode = exception.HttpStatusCode;
+
+                        if ((instructionList.Instructions.Contains("return-503") && statusCode == System.Net.HttpStatusCode.ServiceUnavailable)
+                            || (instructionList.Instructions.Contains("return-400") && statusCode == System.Net.HttpStatusCode.BadRequest)
+                            || (instructionList.Instructions.Contains("return-401") && statusCode == System.Net.HttpStatusCode.Unauthorized))
+                        {
+                            Assert.False(expectSuccess);
+                        }
+                        else
+                        {
+                            throw;
+                        }
+                    }
+                    else
+                    {
+                        throw;
+                    }
                 }
             }
```

Comment:
There is probably a mismatch of braces in try catch finally block causing the build to fail. 

Also, I see there was a local method `InstructionContainsErrorCode` to handle different cases of exception. Do we really need to remove that?